src/main/scala/com/fulcrumgenomics/umi/UmiConsensusCaller.scala

```diff
@@ -121,6 +121,27 @@ object UmiConsensusCaller {
     outHeader.addComment(s"Read group ${rg.getId} contains consensus reads generated from ${oldRgs.size} input read groups.")
     outHeader
   }
+
+  /**
+    * Helper method to check that the input BAM is in the correct order for consensus calling.
+    *
+    * @param header the header of the BAM file to be used for consensus calling
+    * @param source a path or string representing the source of the header
+    * @param warn a function to be called when any warnings are detected/emitted
+    * @param error a function to be called when any errors are encountered
+    */
+  def checkSortOrder(header: SAMFileHeader, source: Any, warn: String => Unit, error: String => Unit): Unit = {
```

Comment:
Could you add a return value (`Boolean`)?  While you pass in handlers (`warn`/`error`) for some cases, it would just be nice to have a return value that returns `true` when it _definitely_ is in the correct sort order.

Reply:
Also, what do you think defaulting `warn` and `error`:  `warn: String => Unit = logger.warn, error: String => Unit = fail`?  You'd need to mix in `LazyLogging` into the `object` if you'd agree.

Reply:
I don't like returning Boolean because it's really three-state: ok, probably-ok and definitely-not-ok.  I honestly thought about just copy/pasting the code between the two consensus caller CLPs and creating a super-trait for the two of them, but it all seemed a bit over-engineered to get re-use of ~5 lines of code.

On the defaults ... `fail()` is a clp/FgBioMain thing, so it's not readily available in non CLP classes :/